torch/_dynamo/variables/builtin.py

```diff
@@ -1512,3 +1512,23 @@ def call_not_(self, tx, a):
     call_ne = _comparison
     call_is_ = _comparison
     call_is_not = _comparison
+
+    def call_all(self, tx, it):
+        def all_impl(iterator):
+            for elem in iterator:
+                if not bool(elem):
+                    return False
+                return True
+
+        if isinstance(it, ListIteratorVariable):
+            try:
+                # Constant fold
+                return ConstantVariable.create(all(it.as_python_constant()))
+            except NotImplementedError:
+                return SymNodeVariable.create(
+                    tx,
+                    tx.output.create_proxy(
+                        "call_function", all_impl, *proxy_args_kwargs([it], {})
```

Comment:
Even better
```suggestion
                        "call_function", all, *proxy_args_kwargs([it], {})
```
and remove `all_impl`?

Reply:
We are lowering from the bytecode level to FX-graph level, so no recursion here :)